### PR TITLE
Feature/multiple drag

### PR DESCRIPTION
**Pull request by tkomstadius**
Drop several files at the same time and save them.

Show the files that are images, only jpeg and png at this time.

Also, don't read the files that have no type.

### CODE REVIEW

File: db/schema.rb [hunk 1]
```diff
@@ -81,7 +81,6 @@
     t.datetime "confirmation_sent_at"
     t.string   "unconfirmed_email"
     t.boolean  "own_password",           default: true
-    t.integer  "default_identity_id"
```
Comment (danielronnkvist) [inline db/schema.rb:84]: Ooops, we need this one!

File: app/assets/javascripts/directives/bDropzone.coffee [hunk 2]
```diff
@@ -35,23 +35,34 @@
...
-          scope.object.data = reader.result.split(",")[1]
-          console.log scope.object
-          scope.drop = true
-          if file.type in ['image/jpeg', 'image/png']
-            scope.image = evt.target.result
+      files = event.originalEvent.dataTransfer.files
+      i = 0
+      temp = []
+      while i < files.length
+        addFile = (file) ->
```
Comment (danielronnkvist) [inline app/assets/javascripts/directives/bDropzone.coffee]: It is better to initiate the function outside of the loop.

Comment (tkomstadius) [inline app/assets/javascripts/directives/bDropzone.coffee]: But I call the function within the loop? I read that it has to be closed up to handle multiple files.

**Comment by tkomstadius** [comment]
Hold off with this PR. I'm checking into the possibility of dropping dirs.

File: app/assets/javascripts/directives/bDropzone.coffee (hunk 2 above)
Comment (tkomstadius) [inline app/assets/javascripts/directives/bDropzone.coffee]: ahh ok, I see what you mean.

**Comment by tkomstadius** [comment]
I think this is it! Be very critical please because I have stared myself blind on this code.

**Comment by danielronnkvist** [comment]
Works great! A few minor fixes, feature requests and wonderings:

**1**
I don't think it's a good idea to do all uploads in a single post request. It's better to do one at the time, and by doing so the user could be getting some visual feedback on how the file upload is going by using [this](http://chieffancypants.github.io/angular-loading-bar/) for example. For each example:

``` coffee
_.each $scope.droppedFiles, (file) ->
  $http.post('/upload_drop.json', {object: file, location: location}.then((response) ->
    ...
```

Perhaps the visual feedback can be another branch so we can get this one merged.

**2**
What does it mean when a file has no type?

**3**
The request right now is blocking users from navigating away from the site and or cancel the request. Maybe a way of cancelling the request is needed, both server and client side. I found [this article](http://odetocode.com/blogs/scott/archive/2014/04/24/canceling-http-requests-in-angularjs.aspx) about canceling $http requests. But this could also be moved into another feature branch.

**4**
A tip, the file upload becomes much faster if the whole base64 isn't logged in the console. To skip logging  add `:objects` to the array in _config/initializers/filter_parameter_logging.rb_.

**Comment by klaseskilson** [comment]
Quick comment regarding the progress bar! We already have [this](https://github.com/caarlos0/nprogress-rails) in our gemfile!

File: app/assets/javascripts/directives/bDropzone.coffee [hunk 3]
```diff
@@ -23,7 +25,75 @@
...
+        reader.readAsDataURL file
+      else
+        scope.$apply ->
+          scope.noType = scope.noType + file.name + " has no type and can not be saved"
+      return false
+
+    # handle files from directory?
+    # does not support wierder types such as .cpp and .xcf
+    addFileFromDir = (entry) ->
+      entry.file((file)->
```
Comment (danielronnkvist) [inline app/assets/javascripts/directives/bDropzone.coffee]: Couldn't this call the _addFile_ function? Like this:

``` javascript
entry.file(addFile)
```

Comment (tkomstadius) [inline app/assets/javascripts/directives/bDropzone.coffee]: I don't really know how it works but for every entry there is a built in funtion file to retrieve the file from the entry and not the same as addFile.

**Comment by danielronnkvist** [comment]
Looks good! `DragDropCtrl.coffee` have one extra indentation which can be removed.

**Comment by danielronnkvist** [comment]
Ready for merge :+1:

**Comment by klaseskilson** [comment]
Veri nice!

**Pull request merged**